src/pretix/control/views/item.py

```diff
@@ -1497,6 +1506,11 @@ def form_valid(self, form):
                     'bundles', 'bundles', 'base_item', order=False,
                     serializer=ItemBundleSerializer
                 )
+            elif k == 'program_times':
+                self.save_formset(
+                    'program_times', 'program_times', 'item', order=False,
+                    serializer=ItemProgramTimeSerializer
+                )
```

Comment:
Instead of invalidating the cache in save_formset (which is used for several formsets here), I would just check this formset whether any form in there is either deleted and its instance has a pk or has_changed(). Also checking for change_data makes sure, we do not invalidate_cache twice, when the cache was invalidated anyways due to changes on the item.

```suggestion
                )
                if not change_data:
                    for form in v.forms:
                        if (form in v.deleted_forms and form.instance.pk) or form.has_changed():
                            invalidate_cache.apply_async(kwargs={'event': self.request.event.pk, 'item': self.object.pk})
                            break
```

Reply:
this was my first idea, that I threw away after thinking about it too much, but you actually made it work (and better). thank you!